mollie_subscription_ept/models/payment_transaction.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from odoo import _, models
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _get_transaction_customer_id(self):
+        mollie_customer_id = False
+        partner_obj = False
+        if self.sale_order_ids and self.sale_order_ids[0]:
+            order_source = self.sale_order_ids[0]
+            partner_obj = order_source.partner_id
+        if partner_obj and partner_obj.mollie_customer_id:
+            mollie_customer_id = partner_obj.mollie_customer_id
+        elif partner_obj:
+            customer_id_data = self.acquirer_id._api_mollie_create_customer_id(partner_obj)
+            if customer_id_data and customer_id_data.get('id'):
+                mollie_customer_id = customer_id_data.get('id')
+                partner_obj.write({'mollie_customer_id': mollie_customer_id})
+        return mollie_customer_id
```

Comment:
Can be modified as 
    def _get_transaction_customer_id(self):
        mollie_customer_id = False
        if self.sale_order_ids:
            partner_obj = self.sale_order_ids[0].partner_id
            if partner_obj.mollie_customer_id:
                mollie_customer_id = partner_obj.mollie_customer_id
            else:
                customer_id_data = self.acquirer_id._api_mollie_create_customer_id(partner_obj)
                if customer_id_data and customer_id_data.get('id'):
                    mollie_customer_id = customer_id_data.get('id')
                    partner_obj.write({'mollie_customer_id': mollie_customer_id})
        return mollie_customer_id